Common/Securities/SecurityCacheProvider.cs

```diff
@@ -0,0 +1,98 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+
+namespace QuantConnect.Securities
+{
+    /// <summary>
+    /// A helper class that will provide <see cref="SecurityCache"/> instances
+    /// </summary>
+    /// <remarks>The value of this class and its logic is performance.
+    /// This class allows for two different <see cref="Security"/> to share the same
+    /// data type cache through different instances of <see cref="SecurityCache"/>.
+    /// This is used to directly access custom data types through their underlying</remarks>
+    public class SecurityCacheProvider
+    {
+        private readonly ConcurrentDictionary<SecurityIdentifier, ConcurrentBag<Symbol>> _relatedSymbols;
```

Comment:
not sure if there's a difference between using a `Symbol` vs `SecurityIdentifier` as a dictionary key -- I think `Symbol` delegates it's `GetHashCode` and `Equals` methods to `SecurityIdentifier`.

Reply:
True, I'll use `Symbol` for consistency with the rest of Lean 👍